### PR TITLE
Add Test Explorer FAQ

### PR DESCRIPTION
**Pull request by kendrahavens**
Add Test Explorer FAQ

This FAQ is needed to explain more details on new test discovery behavior starting in Visual Studio 2017 Update 15.6 Preview 3.

@rpetrusha

### CODE REVIEW

File: docs/test/test-explorer-faq.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+---
+# Visual Studio Test Explorer FAQ
+
+## Test Discovery
+
+### 1. The Test Explorer is not discovering my tests that have theories, custom adapters, custom traits, use #ifdefs, or are dynamically defined. How can I discover these tests?
+
+  Build your project and make sure assembly based discovery is turned on in **Tools > Options > Test**.
+
+  [Real Time Test Discovery](https://go.microsoft.com/fwlink/?linkid=862824), which is source-based test discovery, can’t discover some tests in cases such as theories, custom adapters, custom traits, use #ifdefs, or that are dynamically defined in some other way. A build is required for those tests to be accurately discovered. In the 15.6 Previews assembly-based discovery (the traditional discoverer) will run only after builds. This means Real Time Test Discovery will discover as many tests as it can as you are typing in between builds, but still allow theories (or any dynamically defined tests) to appear after a build. This will improve the responsiveness but still allow you to get complete and accurate results after a build.
```
Comment (rpetrusha) [inline docs/test/test-explorer-faq.md]: - It's clearer to rephrase the first sentence somewhat: ...can't discover tests that use theories, custom adapters, custom traits, `#ifdef` statements, or that are dynamically defined in some other way."
- comma after Previews
- will run only --> runs only (in general, it's best to use present tense rather than future)
- but still allow: real-time test discovery doesn't allow. Possibly rephrase this to "in between builds, and assembly-based discovery finds theories (or any dynamically defined tests) that appear after the build. 
- "This will improve responsiveness" --> It's not clear what "this" is
- the performance -> performance

File: docs/test/test-explorer-faq.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+### 2. What does the '+' (plus) symbol that appears in the top line of the Test Explorer mean?
+
+  The '+' (plus) symbol indicates more tests may be discovered on build if discovering tests from built assemblies is on. This will appear if dynamically defined tests are detected in your project.
```
Comment (rpetrusha) [inline docs/test/test-explorer-faq.md]: Perhaps phrase this in a way that's consistent with the previous item: "...more tests may be discovered after a build is assembly-based discovery is turned on."

File: docs/test/test-explorer-faq.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  ![Plus symbol summary line](media/testex-plussymbol.png)
+
+### 3. Assembly based discovery is no longer working for my project. How do I turn it back on?
+
+  Go to **Tools > Options > Test** and check the box for 'Additionally discover tests built from assemblies after builds.'
```
Comment (rpetrusha) [inline docs/test/test-explorer-faq.md]: Use bold rather than fencing here; fenced text won't be localized.

File: docs/test/test-explorer-faq.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+  ![Assembly based option](media/testex-toolsoptions.png)
+
+### 4. Tests now appear in the Test Explorer as I type without having to build my project. What changed?
+
+  This feature is called [Real Time Test Discovery](https://go.microsoft.com/fwlink/?linkid=862824). It uses a Roslyn analyzer to discover tests and populate the test explorer in real time without requiring you to build your project. See FAQ #1 for more info on test discovery behavior on dynamically defined tests such as theories or custom traits.
```
Comment (rpetrusha) [inline docs/test/test-explorer-faq.md]: test explorer --> Test Explorer
on dynamically --> for dynamically

File: docs/test/test-explorer-faq.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+### 5. What languages and test frameworks can use Real Time Test Discovery?
+
+  [Real Time Test Discovery](https://go.microsoft.com/fwlink/?linkid=862824) only works for the managed languages, C# and Visual Basic since it is built using the Roslyn compiler. For now, Real Time Test Discovery only works for the xUnit, NUnit, and MSTest frameworks.
```
Comment (rpetrusha) [inline docs/test/test-explorer-faq.md]: no comma after languages

File: docs/test/test-explorer-faq.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+## Features
+
+### How can I turn on feature flags to try out new testing features?
+
+Feature flags are used to ship **experimental or unfinished** parts of the product to avid users who would like to give feedback before the features have officially shipped. They may destabilize your IDE experience. It is recommended to use them only on safe dev environments such as VMs. These are always use at your own risk settings. You can turn on **experimental** features with the [Feature Flags extension](https://marketplace.visualstudio.com/items?itemName=PaulHarrington.FeatureFlagsExtension) or through the developer command prompt.
+
+![Feature Flag Extension](media/testex-featureflag.png)
+
+If you would like to turn on a feature flag through the Visual Studio developer command prompt use the following command. You must change the path to where Visual Studio is installed on your machine and change the registry key to the desired feature flag.
```
Comment (rpetrusha) [inline docs/test/test-explorer-faq.md]: comma after prompt

File: docs/test/test-explorer-faq.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+```
+vsregedit set “C:\Program Files (x86)\Microsoft Visual Studio\Preview\Enterprise” HKLM FeatureFlags\TestingTools\UnitTesting\HierarchyView Value dword 1
+```
+
+> [!NOTE] You can turn off the flag with the same command by using a 0 instead of a 1 after dword.
+  
+## See Also  
+ <xref:Microsoft.VisualStudio.TestTools.UnitTesting>
```
Comment (rpetrusha) [inline docs/test/test-explorer-faq.md]: I think the link text appears as just "UnitTesting". To fully qualify it, "\<xref:Microsoft.VisualStudio.TestTools.UnitTesting?displayProperty=fullName>"

**Comment by gewarren** [comment]
Kendra created this PR in the private repo instead.
